From: Ben Greear <greearb@candelatech.com>
Date: Fri, 1 Apr 2016 14:12:08 -0700
Subject: [PATCH] ath10k: Ensure txrx-compl-task is stopped when cleaning
 htt-tx.

Otherwise, the txrx-compl-task may access some bad memory?

Signed-off-by: Ben Greear <greearb@candelatech.com>
---

--- a/drivers/net/wireless/ath/ath10k/htt_tx.c
+++ b/drivers/net/wireless/ath/ath10k/htt_tx.c
@@ -388,6 +388,8 @@ void ath10k_htt_tx_free(struct ath10k_ht
 {
 	int size;
 
+	tasklet_kill(&htt->txrx_compl_task);
+
 	idr_for_each(&htt->pending_tx, ath10k_htt_tx_clean_up_pending, htt->ar);
 	idr_destroy(&htt->pending_tx);
 
